code/game/objects/structures/mirror.dm

```diff
@@ -281,7 +285,30 @@
 				H.dna.update_ui_block(DNA_EYE_COLOR_BLOCK)
 				H.update_body()
 	if(choice)
-		curse(user)
-
-/obj/structure/mirror/magic/proc/curse(mob/living/user)
+		after_use(user)
+
+/**
+ * Called after a given user has used the magic mirror in some manner.
+ *
+ * Subclasses use this generally to do horrible things to the user for
+ * daring to be vain.
+ */
+/obj/structure/mirror/magic/proc/after_use(mob/living/user)
 	return
+
+/obj/structure/mirror/magic/lesser
+	desc = "Turn and face the strange, yet familiar face."
+
+/obj/structure/mirror/magic/lesser/setup_choosable_races()
+	choosable_races = sortList(GLOB.roundstart_races)
+
+/obj/structure/mirror/magic/badmin
+	desc = "Turn and face, for you have power without grace."
+
+/obj/structure/mirror/magic/badmin/setup_choosable_races()
+	for(var/speciestype in subtypesof(/datum/species))
+		var/datum/species/S = speciestype
+		if(initial(S.changesource_flags) & MIRROR_BADMIN)
+			choosable_races += initial(S.id)
```

Comment:
```suggestion
	for(var/datum/species/speciestype as anything in subtypesof(/datum/species))
		if(initial(speciestype.changesource_flags) & MIRROR_BADMIN)
			choosable_races += initial(speciestype.id)
```
* Better var names.
* `as anything` to avoid type casting.